### ISSUE TITLE
regex

### ISSUE THREAD

**Issue by Kinomajiro** [Kinomajiro/235-Lab-01#1]
whenever i build it tells me that the regex and the smatch are not defined in scope

**Comment by Kinomajiro** [comment]
specifically it has issues with

std::smatch m;
std::regex info ("([:w:])\s(A|C|R)\s(\d+)\s(\d+)\s(\d+)\s(\d+)");
std::regex_search(fighter, m, info);

from archer.cpp

**Comment by kupiakos** [comment]
I've noticed the regular expression strings in [Archer.cpp](https://github.com/Kinomajiro/235-Lab-01/blob/master/Archer.cpp#L7), [Cleric.cpp](https://github.com/Kinomajiro/235-Lab-01/blob/master/Cleric.cpp#L7), [Robot.cpp](https://github.com/Kinomajiro/235-Lab-01/blob/master/Robot.cpp), and [Roster.cpp](https://github.com/Kinomajiro/235-Lab-01/blob/master/Roster.cpp#L7) do not escape the backslashes. Notice it highlights the problems in red. That might be part of the problem.

You probably want something more like

``` c++
regex info("(\\w+) (A|C|R) (\\d+) (\\d+) (\\d+) (\\d+)");
```

Also, I noticed there's a lot of code duplication. You can probably do all of this parsing in one place.

Read [the page on Github Markdown](https://guides.github.com/features/mastering-markdown/) to learn how to format your comments things like code and such.

[Here](http://en.cppreference.com/w/cpp/regex/ecmascript) is the page on the grammar used by the C++ regex library.

**Comment by Kinomajiro** [comment]
yeah i could but that would add more files and by the time i got around to it it was like a ton of extra work

**Comment by Kinomajiro** [comment]
and i fixed the spaces but not the escape characters

**Comment by kupiakos** [comment]
It should _reduce_ the complexity of your project, not increase it. What do you mean by "fixed the spaces but not the escape characters"?

**Comment by Kinomajiro** [comment]
i removed the /s from in the middle of the groups

**Comment by kupiakos** [comment]
Try pushing your new changes for practice.

With regards to "not defined in scope", are you sure you're compiling with C++11 support enabled?

**Comment by Kinomajiro** [comment]
i put the -std=c++11 flag in the compiler

**Comment by Kinomajiro** [comment]
/home/austin/Pictures/Screenshot from 2015-09-22 18:47:02.png

**Comment by Kinomajiro** [comment]
trying to see if i can post an image of my problem

**Comment by kupiakos** [comment]
Also, you should not be using `#pragma once` in your `.cpp` files.

**Comment by kupiakos** [comment]
Push all changes you make so I can see them.

You can do that with

``` sh
$ git commit -a -m 'My commit message'
$ git push
```

**Comment by Kinomajiro** [comment]
just updated the files

**Comment by kupiakos** [comment]
Still have `#pragma once` at the top of your `.cpp` files. Also, I'm seeing errors with your retrieving of the values from the regex, such as [here](https://github.com/Kinomajiro/235-Lab-01/blob/master/Archer.cpp#L11-L18). Indexing the match results returns an `std::string`. You probably want to use `atoi` or similar to get the results as an `int`. (and maybe `std::cmatch` instead of `std::smatch`).

**Comment by kupiakos** [comment]
You should move _all_ of this logic to `Roster::addFighter` though. Change the constructors for `Archer`, etc. to take all of the parameters required instead of all of this code duplication. Parsing should be done before instantiating your `Fighter`s.

**Comment by Kinomajiro** [comment]
kk i'll do that it would limit my errors to one place instead of several

**Comment by Kinomajiro** [comment]
i uploaded a new version

**Comment by Kinomajiro** [comment]
new error

Description Resource    Path    Location    Type
#error This file requires compiler and library support for the ISO C++ 2011 standard. This support is currently experimental, and must be enabled with the -std=c++11 or -std=gnu++11 compiler options. lab_1-RPG       line 32, external location: /usr/include/c++/4.8/bits/c++0x_warning.h   C/C++ Problem

**Comment by kupiakos** [comment]
- `Roster.cpp` still has `#pragma once`.
- You're missing a semicolon [here](https://github.com/Kinomajiro/235-Lab-01/blob/master/Robot.cpp#L90).
- You should probably name the parameters `fighter` and `Class` in your fighter constructors `name` and `className` for consistency.
- You should probably just be creating local variables storing the fighter information around [here](https://github.com/Kinomajiro/235-Lab-01/blob/master/Roster.cpp#L16), **doing the proper conversions to `int` first**.
- Based on that error message, it's clear that C++ 11 support is not enabled.
- You should be surrounding your blocks of code with three backticks (```).

**Comment by Kinomajiro** [comment]
i updated the files with your changes

**Comment by kupiakos** [comment]
- `Roster.cpp` _still_ has `#pragma once`.
- You need to be surrounding the assignment on [these lines](https://github.com/Kinomajiro/235-Lab-01/blob/c5dffcba08b4be670cb8486ddfcd824dcfb29ff8/Roster.cpp#L20-L23) with [`atoi`](http://en.cppreference.com/w/cpp/string/byte/atoi) as previously described. 
- You should probably be comparing with a string on [this line](https://github.com/Kinomajiro/235-Lab-01/blob/c5dffcba08b4be670cb8486ddfcd824dcfb29ff8/Roster.cpp#L25) and [this line](https://github.com/Kinomajiro/235-Lab-01/blob/c5dffcba08b4be670cb8486ddfcd824dcfb29ff8/Roster.cpp#L30).

**Comment by Kinomajiro** [comment]
updated the files

**Comment by kupiakos** [comment]
I only see two more problems, both of them linking errors:
- No main function is defined. This is expected with the testing driver.
- Since you declared `Roster::Roster()` in [`Roster.h`](https://github.com/Kinomajiro/235-Lab-01/blob/c5dffcba08b4be670cb8486ddfcd824dcfb29ff8/Roster.h#L15) but did not implement it in `Roster.cpp`, it can't link against it. I'd recommend deleting the declaration in `Roster.h`.

If you are currently having additional errors, it's because your environment is not set up to compile C++ 11.
